commitment/split.go

```diff
@@ -133,6 +151,20 @@ func NewSplitCommitment(input *asset.Asset, outPoint wire.OutPoint,
 		return nil, ErrInvalidSplitLocator
 	}
 
+	// The only valid unspendable root locator uses the correct unspendable
+	// script key and has zero value.
+	if rootLocator.Amount == 0 &&
+		rootLocator.ScriptKey != asset.NScriptKey {
+
+		return nil, ErrInvalidScriptKey
+	}
+
+	if rootLocator.Amount != 0 &&
+		rootLocator.ScriptKey == asset.NScriptKey {
```

Comment:
Don't need to do it here rn, but it might make sense to add this check in a few other places when we interact w/ assets. Similar to what we do at times when we ensure that an asset doesn't have a genesis witness, etc, etc.

Reply:
Agree, the only reason not to use some version of `IsUnspendable()` here is to give a more specific error - I think just using that will be sufficient elsewhere.